impl/maven-core/src/main/java/org/apache/maven/internal/transformation/impl/PomInlinerTransformer.java

```diff
@@ -52,35 +61,79 @@ class PomInlinerTransformer extends TransformerSupport {
         this.interpolator = requireNonNull(interpolator);
     }
 
+    @Override
+    public InstallRequest remapInstallArtifacts(RepositorySystemSession session, InstallRequest request) {
+        return request.setArtifacts(replacePom(session, request.getArtifacts()));
+    }
+
+    @Override
+    public DeployRequest remapDeployArtifacts(RepositorySystemSession session, DeployRequest request) {
+        return request.setArtifacts(replacePom(session, request.getArtifacts()));
+    }
+
+    private Collection<Artifact> replacePom(RepositorySystemSession session, Collection<Artifact> artifacts) {
+        Set<String> needsInlining = needsInlining(session);
+        if (needsInlining.isEmpty()) {
+            return artifacts;
+        }
+        ArrayList<Artifact> newArtifacts = new ArrayList<>(artifacts.size());
+        for (Artifact artifact : artifacts) {
```

Comment:
```suggestion
        return artifacts.map(....)
```
give special attention avoiding overhead giving only important stuff which is for body as `map` operation as well. Its already very nice functional and very compact. Lets embrace here as well. 

its just 1 LOC 

`return isEmpty() ? artifacts : artifacts.map(toFoo()) `